### PR TITLE
Fix Datastream private connectivity docs

### PR DESCRIPTION
**Pull request by sagiyosefia**
Fix a bug with Datastream private connectivity docs, as bigquery destination doesn't supports private connection config. Changing the example to work with Postgresql instead
b/307440603

```release-note:none
```

### CODE REVIEW

**Review by dani29 — APPROVED** [review]

**Review by rileykarson — COMMENTED** [review]
AH- we'll want the sample to succeed, so we'll want to connect to an actual instance. Can we add a SQL instance that's configured to accept a connection easily?

Error message:

> We can't connect to the source database using the specified hostname and port. Make sure that the hostname and port are correct and that the database can accept inbound connections.

File: mmv1/products/datastream/ConnectionProfile.yaml [hunk 1]
```diff
@@ -36,12 +36,16 @@ examples:
...
-    name: 'datastream_connection_profile_bigquery_private_connection'
+    name: 'datastream_connection_profile_postgresql_private_connection'
     primary_resource_id: 'default'
     vars:
       private_connection_id: 'my-connection'
       connection_profile_id: 'my-profile'
       network_name: 'my-network'
+      database_instance_name: 'my-instance'
+      deletion_protection:
+        'true'
```
Comment (rileykarson) [inline mmv1/products/datastream/ConnectionProfile.yaml:47]: We'll need to set var overrides to `false` so the test / oics can succeed in deletion: https://github.com/GoogleCloudPlatform/magic-modules/blob/905133647a32023e4aab64b0336b3f265f563295/mmv1/products/sql/Database.yaml#L42C1-L45

That will put `true` in the docs, and `false` in other spots.

Comment (sagiyosefia) [reply mmv1/products/datastream/ConnectionProfile.yaml]: done

File: mmv1/products/datastream/ConnectionProfile.yaml [hunk 2]
```diff
@@ -36,12 +36,18 @@ examples:
     vars:
       connection_profile_id: 'my-profile'
   - !ruby/object:Provider::Terraform::Examples
-    name: 'datastream_connection_profile_bigquery_private_connection'
+    name: 'datastream_connection_profile_postgresql_private_connection'
```
Comment (rileykarson) [inline mmv1/products/datastream/ConnectionProfile.yaml:39]: This will need `pull_external: true` to use the random provider!

**Review by rileykarson — COMMENTED** [review]
New error- these should be happening locally too, I'd recommend running there to save round trips.

File: mmv1/products/datastream/ConnectionProfile.yaml (hunk 2 above)
Comment (sagiyosefia) [reply mmv1/products/datastream/ConnectionProfile.yaml:39]: QQ: When the tests are triggered? I saw it's triggered long time after a new commit is added

**Comment by sagiyosefia** [comment]
QQ: When the tests are triggered? I saw it's triggered long time after a new commit is added

**Comment by rileykarson** [comment]
Manually right now- there are instructions in our internal contribution guide to enable autorun for yourself (we need to check public information because we're checking in public code, so there's an extra step)

**Comment by sagiyosefia** [comment]
@rileykarson can you please approve the PR, the tests are passing. Thanks

**Review by rileykarson — APPROVED** [review]

**Pull request merged**